app/src/stores/post.ts

```diff
@@ -42,7 +43,22 @@ export const usePostStore = defineStore("post", () => {
             return new Promise((resolve) => {
                 resolve(
                     posts.filter((post) => {
-                        return post.content[0]?.status == ContentStatus.Published;
+                        const isPublished = post.content[0]?.status == ContentStatus.Published;
+                        const publishDate = post.content[0]?.publishDate
+                            ? post.content[0]?.publishDate
+                            : null;
+                        const expiryDate = post.content[0]?.expiryDate
+                            ? post.content[0]?.expiryDate
+                            : null;
+                        const isNotExpired =
+                            !post.content[0]?.expiryDate ||
+                            post.content[0]?.expiryDate > DateTime.now();
+
+                        const isAvailable =
+                            (publishDate ? publishDate <= DateTime.now() : true) &&
+                            (expiryDate ? expiryDate > DateTime.now() : true);
```

Comment:
I notice two things here:
- `isAvailable` also checks for expiry date, which is double with `isNotExpired` above, so doesn't seem needed to me.
- This can be answered by Ivan, but not sure if it makes sense to check publish date at all? Or is that how we'll handle future published, by setting `status` to `published`, and having a publish date in the future?

Reply:
Yes @laurensgroeneveld ! He said that we can schedule posts in the CMS for the future. For that, i think we need to look at the both `publishDate` and `expiryDate`